engine/swarm/services.md

```diff
@@ -507,41 +531,55 @@ service based upon node attributes and metadata or engine metadata. For more
 information on constraints, refer to the `docker service create`
 [CLI reference](/engine/reference/commandline/service_create.md).
 
-Use placement preferences to divide tasks evenly over different categories of
-nodes. An example of where this may be useful is balancing tasks between
-multiple datacenters or availability zones. In this case, you can use a
-placement preference to spread out tasks to multiple datacenters and make the
-service more resilient in the face of a localized outage. You can use
-additional placement preferences to further divide tasks over groups of nodes.
-For example, you can balance them over multiple racks within each datacenter.
-For more information on constraints, refer to the `docker service create`
-[CLI reference](/engine/reference/commandline/service_create.md).
 
-### Reserve memory or CPUs for a service
+#### Placement constraints
 
-To reserve a given amount of memory or number of CPUs for a service, use the
-`--reserve-memory` or `--reserve-cpu` flags. If no available nodes can satisfy
-the requirement (for instance, if you request 4 CPUs and no node in the swarm
-has 4 CPUs), the service remains in a pending state until a node is available to
-run its tasks.
+Use placement constraints to control the nodes a service can be assigned to. For
+instance, you might want to divide tasks evenly over different categories of
+nodes, using arbitrary metadata that you apply to the nodes. If no
+appropriately-labelled nodes are available, the service will not run. This
+differentiates placement constraints from
+[placement preferences](#placement-preferences).
 
-#### Out Of Memory Exceptions (OOME)
+An example of where placement constraints may be useful is balancing tasks
+between multiple datacenters or availability zones. In this case, you can use a
+placement constraint to spread out tasks to multiple datacenters and make the
+service more resilient in the face of a localized outage. You can use additional
+placement constraint to further divide tasks over groups of nodes. For example,
+you can balance them over multiple racks within each datacenter.
 
-If your service attempts to use more memory than the swarm node has available,
-you may experience an Out Of Memory Exception (OOME) and a container, or the
-Docker daemon, might be killed by the kernel OOM killer. To prevent this from
-happening, ensure that your application runs on hosts with adequate memory and
-see
-[Understand the risks of running out of memory](/engine/admin/resource_constraints.md#understand-the-risks-of-running-out-of-memory).
+In the following example, you constrain a service to only run on nodes with
+the [label](engine/swarm/manage-nodes.md#add-or-remove-label-metadata) `region`
+set to `east`, using `--constraint` flag with an equality operator
+(`==` or `!=`). You can also use the `constraint` key in a `docker-compose.yml`
```

Comment:
perhaps link to the constraint key in the docker-compose file reference for discoverability?